### PR TITLE
feat: V1 endpoints for friendship support

### PR DESCRIPTION
**Pull request by jmoguilevsky**
This PR adds all necessary endpoints for M1:
- Friendships API
- Configuration for privacy

Friendship requests will be created using matrix events so for the moment they don't need an endpoint

### CODE REVIEW

**Review by lauti7 — APPROVED** [review]

File: openapi/components/examples/friends/200-get-friends-count.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "value": 10
```
Comment (agusaldasoro) [inline openapi/components/examples/friends/200-get-friends-count.json]: Shouldn't this be a JSON?
```suggestion
  "value":  { "count": 10 }
```

File: openapi/components/examples/friends/200-get-account-configuration.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "value": {
+    "friends": "FRIENDS_OF_FRIENDS",
```
Comment (agusaldasoro) [inline openapi/components/examples/friends/200-get-account-configuration.json]: What's this? A string?

File: openapi/components/examples/friends/200-get-account-configuration.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+    "friendsCount": "EVERYONE"
+  }
```
Comment (agusaldasoro) [inline openapi/components/examples/friends/200-get-account-configuration.json]:
```suggestion
  "value": { 
  "config: {
  
    "friends": "FRIENDS_OF_FRIENDS",
    "friendsCount": "EVERYONE"
  }}
```

File: openapi/components/examples/friends/200-set-account-configuration.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "value": {
```
Comment (agusaldasoro) [inline openapi/components/examples/friends/200-set-account-configuration.json:2]: I'd add a first field named `config` here too

File: openapi/components/examples/friends/200-get-own-friend-requests.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "value": {
+    "incoming": [
+      "0xad0a8680845B19f833e317F00CCE41967c85253a",
```
Comment (agusaldasoro) [inline openapi/components/examples/friends/200-get-own-friend-requests.json]: what about transforming each item into an object? Maybe you need more information, for example the timestamp.

File: openapi/components/examples/friends/200-get-account-configuration.json (hunk 2 above)
Comment (jmoguilevsky) [reply openapi/components/examples/friends/200-get-account-configuration.json]: yes

**Review by agusaldasoro — APPROVED** [review]
Great work!!

**Pull request merged**